packages/teams-js/test/public/video.spec.ts

```diff
@@ -23,38 +23,426 @@ describe('video', () => {
       app._uninitialize();
     }
   });
+  describe('registerForVideoFrame', () => {
+    let emptyVideoEffectCallback = (
+      _frame: video.VideoFrame,
+      _notifyVideoFrameProcessed: () => void,
+      _notifyError: (errorMessage: string) => void,
+    ): void => {};
+    const videoFrameConfig: video.VideoFrameConfig = {
+      format: video.VideoFrameFormat.NV12,
+    };
+
+    const allowedContexts = [FrameContexts.sidePanel];
+    Object.values(FrameContexts).forEach(context => {
+      if (allowedContexts.some(allowedContext => allowedContext === context)) {
+        return;
+      }
+
+      it('FRAMED - should not allow registerForVideoFrame calls from the wrong context', async () => {
+        await framedPlatformMock.initializeWithContext(context);
+
+        expect(() => video.registerForVideoFrame(emptyVideoEffectCallback, videoFrameConfig)).toThrowError(
+          `This call is only allowed in following contexts: ${JSON.stringify(
+            allowedContexts,
+          )}. Current context: "${context}".`,
+        );
+      });
+
+      it('FRAMELESS - should not allow registerForVideoFrame calls from the wrong context', async () => {
+        await framelessPlatformMock.initializeWithContext(context);
+
+        expect(() => video.registerForVideoFrame(emptyVideoEffectCallback, videoFrameConfig)).toThrowError(
+          `This call is only allowed in following contexts: ${JSON.stringify(
+            allowedContexts,
+          )}. Current context: "${context}".`,
+        );
+      });
+    });
+
+    it('FRAMED - should successfully send registerForVideoFrame message', async () => {
+      await framedPlatformMock.initializeWithContext(FrameContexts.sidePanel);
+      video.registerForVideoFrame(emptyVideoEffectCallback, videoFrameConfig);
+      const message = framedPlatformMock.findMessageByFunc('video.registerForVideoFrame');
+      expect(message).not.toBeNull();
+      expect(message.args.length).toBe(1);
+      expect(message.args).toEqual([videoFrameConfig]);
+    });
+
+    it('FRAMELESS - should successfully send registerForVideoFrame message', async () => {
+      await framelessPlatformMock.initializeWithContext(FrameContexts.sidePanel);
+      video.registerForVideoFrame(emptyVideoEffectCallback, videoFrameConfig);
+      const message = framelessPlatformMock.findMessageByFunc('video.registerForVideoFrame');
+      expect(message).not.toBeNull();
+      expect(message.args.length).toBe(1);
+      expect(message.args).toEqual([videoFrameConfig]);
+    });
+
+    it('FRAMED - should successful register video frame handler', async () => {
```

Comment:
Nit: typo 'successful'. Saw this a few times in other parts of the file too.

Reply:
Thanks for catching. Will fix it.